clients/fides-js/src/fides-tcf.ts

```diff
@@ -202,22 +204,31 @@ const updateCookieAndExperience = async ({
 };
 
 /**
- * If a fidesString is explicitly passed in, we override the associated cookie props, which are then
- * used to override associated props in the experience.
+ * If a fidesString is provided either explicitly or retrieved with a custom get preferences fn,
+ * we override the associated cookie props, which are then used to override associated props in the experience.
  */
 const updateFidesCookieFromString = (
   cookie: FidesCookie,
   fidesString: string,
-  debug: boolean
+  debug: boolean,
+  fidesStringVersionHash: string | undefined
 ): { cookie: FidesCookie; success: boolean } => {
   debugLog(
     debug,
     "Explicit fidesString detected. Proceeding to override all TCF preferences with given fidesString"
   );
   try {
     const cookieKeys = transformFidesStringToCookieKeys(fidesString, debug);
+    const updatedCookie = {
+      ...cookie,
+      tcf_consent: cookieKeys,
+      fides_string: fidesString,
+    };
+    if (fidesStringVersionHash) {
+      updatedCookie.tcf_version_hash = fidesStringVersionHash;
+    }
     return {
-      cookie: { ...cookie, tcf_consent: cookieKeys, fides_string: fidesString },
+      cookie: updatedCookie,
```

Comment:
rather than this `if`, I think it might be cleaner and work just as well to always set `fidesStringVersionHash`. at worst it's undefined, which is ok since we just compare to the experience version hash which [will be defined when we compare it](https://github.com/ethyca/fides/blob/PROD-1241/clients/fides-js/src/lib/consent-utils.ts/#L247-L249). that makes this bit of code cleaner if we can keep it as...

```typescript
return {
  cookie: {...cookie, tcf_consent: cookieKeys, fides_string: fidesString, tcf_version_hash: fidesStringVersionHash
}
```
mostly thinking of ways to keep fewer branches in our initialization code!

Reply:
hmm won't this potentially overwrite a previously-saved `tcf_version_hash` from the cookie? E.g. scenario: A new user visits a site with Fides. In this site's config, there is no custom get preferences fn configured. The user saves their preferences, which writes the cookie to reflect the user preferences and `tcf_version_hash` from the `experience`. Next time they visit the site, in the init of Fides, their browser cookie's `tcf_version_hash` gets set to `undefined`, since there is no `fidesStringVersionHash`. This causes `shouldResurfaceConsent` to return `true`, which would mean the banner will keep resurfacing for the user.

Is that right, or am I missing something?

I can keep the code more minimal by doing something like this:

```
return {
  cookie: {...cookie, tcf_consent: cookieKeys, fides_string: fidesString, tcf_version_hash: fidesStringVersionHash ?? cookie.tcf_version_hash
}
```

Reply:
oh yeah, that's a good point! I think your suggested change makes sense 👍